src/gen9_vdenc.c

```diff
@@ -3456,14 +3475,16 @@ gen9_vdenc_mfx_vdenc_avc_slices(VADriverContextP ctx,
                     insert_mi_flush = 0;
                 }
 
+                pipeline_flush_params.mfx_pipeline_done = 1;
                 pipeline_flush_params.vdenc_pipeline_done = 1;
                 pipeline_flush_params.vdenc_pipeline_command_flush = 1;
                 pipeline_flush_params.vd_command_message_parser_done = 1;
                 gen9_vdenc_vd_pipeline_flush(ctx, encoder_context, &pipeline_flush_params);
 
                 if (insert_mi_flush) {
                     memset(&mi_flush_dw_params, 0, sizeof(mi_flush_dw_params));
-                    mi_flush_dw_params.video_pipeline_cache_invalidate = 1;
+                    mi_flush_dw_params.video_pipeline_cache_invalidate = 0;
+                    mi_flush_dw_params.bo = vdenc_context->vdenc_internal_res.bo;
                     gen8_gpe_mi_flush_dw(ctx, batch, &mi_flush_dw_params);
```

Comment:
Yes. For the SKL/BXT, only one single slice is supported.
So it is not necessary for SKL.
But for KBL, this is needed for multi-slices.